### ISSUE TITLE
RFC aioespnow asyncio interface

### ISSUE THREAD

**Issue by peterhinch** [glenn20/micropython-espnow-utils#3]
I'm very interested in the way aioespnow works - I don't know if you've looked at [this section](https://github.com/peterhinch/micropython-async/blob/master/v3/docs/TUTORIAL.md#64-writing-streaming-device-drivers) of my asyncio tutorial. The code samples hook into the asyncio I/O mechanism by creating artificial StreamReader and/or StreamWriter instances. While this works, it is not pretty.

aioespnow.py has `yield asyncio.core._io_queue.queue_read(self)` which is very neat. My `MillisecTimer` example can be rewritten as
```python
class MillisecTimer(io.IOBase):
    def __init__(self):
        self.end = 0

    def __call__(self, ms):
        self.end = utime.ticks_add(utime.ticks_ms(), ms)
        return self

    def __iter__(self):
        yield asyncio.core._io_queue.queue_read(self)  # The magic happens here :-)
        
    def ioctl(self, req, arg):
        ret = MP_STREAM_ERROR
        if req == MP_STREAM_POLL:
            ret = 0
            if arg & MP_STREAM_POLL_RD:
                if utime.ticks_diff(utime.ticks_ms(), self.end) >= 0:
                    ret |= MP_STREAM_POLL_RD
        return ret
```
I'm interested in this use of an undocumented interface. Did the maintainers provide any input? I'd like to see an approved interface for this that I could recommend to users.

**Comment by glenn20** [comment]
In earlier versions I had also attempted to overlay the streamreader interface, but went with this approach for simplicity and conciseness. It was concerned that the size of the espnow PR was what was holding it back from getting merged and felt the need to save code wherever I could.

I can't speak for Damien, but I did point it out to him during code review and he responded that he accepted it as it is "hidden" behind a user api and the usage can be managed within the micropython project and there is not a better way to support this in uasyncio at this stage. I'm sure he wouldn't want it recommended for users in their own code as it is an implementation detail that almost certainly will change in future and we can update aioespnow.py when needed.

**Comment by glenn20** [comment]
Oh - and I agree - an approved interface to this capability would be very useful. All you need to do is provide support for `poll()` and your class can be hooked into the asyncio framework.

**Comment by peterhinch** [comment]
Thanks for that insight.

Damien's commitment to CPython compatibility seems to result in a reluctance to accept performance-related extensions. Yet (u)asyncio would be useless in a firmware context without `sleep_ms` etc. When I first got involved (9 years ago) uasyncio V1 had a time resolution limited to 1s. The first thing I did was to write a cooperative scheduler.

I fail to see the problem with a firmware-targeted, CPython-compatible, asyncio superset.
